core/eth/state.go

```diff
@@ -10,10 +10,10 @@ import (
 
 type ChainState struct {
 	Client common.EthClient
-	Tx     core.Transactor
+	Tx     core.Reader
 }
 
-func NewChainState(tx core.Transactor, client common.EthClient) *ChainState {
+func NewChainState(tx core.Reader, client common.EthClient) *ChainState {
```

Comment:
will this be part of `Reader` down the road?

Reply:
I think so. Will be a follow up.